### PR TITLE
Adding in validation for trailing dot at end of DNS record name

### PR DESCRIPTION
**Pull request by NA2047**
Potential fix for issue #9074 I have added a validator to ensure the trailing dot is not missed during terraform plan
https://github.com/hashicorp/terraform-provider-google/issues/9074

If this PR is for Terraform, I acknowledge that I have:

- [ X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dns: Added in validation for trailing dot at end of DNS record name
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Could you add unit tests for this feature?

[ref](https://github.com/GoogleCloudPlatform/magic-modules/blob/147dedb06c6f9542b26a8bc3386c8aede7349ae6/mmv1/third_party/terraform/tests/resource_bigquery_table_test.go#L1287)

**Comment by NA2047** [comment]
> Could you add unit tests for this feature?
> 
> [ref](https://github.com/GoogleCloudPlatform/magic-modules/blob/147dedb06c6f9542b26a8bc3386c8aede7349ae6/mmv1/third_party/terraform/tests/resource_bigquery_table_test.go#L1287)

**Comment by ScottSuarez** [comment]
I notice this is in draft state. Please let me know when it is ready for review

File: mmv1/third_party/terraform/resources/resource_dns_record_set.go [hunk 1]
```diff
@@ -620,3 +621,13 @@ func flattenDnsRecordSetRoutingPolicyGEO(geo *dns.RRSetRoutingPolicyGeoPolicy) [
 	}
 	return ris
 }
+
+func validateRecordNameTrailingDot(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+	last1 := value[len(value)-1:]
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/resources/resource_dns_record_set.go:627]: there is potential to crash here if we are out of bounds. Could you add empty string as a edge case in your test suite?

https://go.dev/play/p/At9a9rVQlRL

Comment (NA2047) [reply mmv1/third_party/terraform/resources/resource_dns_record_set.go:627]: Will do @ScottSuarez

**Comment by ScottSuarez** [comment]
Looks like the linter is failing for your files. Could you run make lint downstream?

**Comment by NA2047** [comment]
> Here is the error from the linter
> 
> ```
> make: *** [fmtcheck] Error 1
> You can use the command: `make fmt` to reformat code.
> ./google-beta/resource_dns_record_set.go
> gofmt needs running on the following files:
> ==> Checking that code complies with gofmt requirements...
> make: *** [fmtcheck] Error 1
> GNUmakefile:23: recipe for target 'fmtcheck' failed
> You can use the command: `make fmt` to reformat code.
> ./google-beta/resource_dns_record_set.go
> gofmt needs running on the following files:
> ==> Checking that code complies with gofmt requirements...
> make: *** No rule to make target 'tools'.  Stop.
> make: *** [fmtcheck] Error 1
> GNUmakefile:23: recipe for target 'fmtcheck' failed
> You can use the command: `make fmt` to reformat code.
> ./google-beta/resource_dns_record_set.go
> gofmt needs running on the following files:
> ==> Checking that code complies with gofmt requirements.
> ```

**Comment by NA2047** [comment]
> looks good

Thanks Scott, I assume you will be merging these in to main then? I don't have the ability to merge

**Comment by ScottSuarez** [comment]
correct, I didn't want to merge until you marked the change ready for review. Merging now.
